### DISCUSSION TITLE
Make Kotlinlang the default selected style.

### DISCUSSION THREAD

**Discussion by JavierSegoviaCordoba** [facebookincubator/ktfmt#446] [Ideas]
As ktfmt is an open source library, I think it should use as default the most similar setup to what Kotlin conventions propose.

That means:
- Use kotlinlang as default in all utilities (Gradle plugins, IDEA plugin, etc).
- Rename default to Meta or Facebook.

> ***Note:***
> I would remove the "default" name which can mislead to what a new user or an old user of ktfmt could expect respectively with the changes and/or previous behavior.

**Comment by ColtonIdle** [comment]
Removing company names would be nice. dropbox, meta, etc.
We should just have a style name for 4 spaces vs 2 and name the style based on the number 🤷

**Reply by JavierSegoviaCordoba** [reply]
There are more than one style with the same indentation but different specifications in other places.

In Java it is the google-java-format is very popular for example.

**Reply by hick209** [reply]
What's wrong with company names?
As long as they are around and have open source code that mostly follows the formatting, I don't really see an issue with it

**Reply by ColtonIdle** [reply]
Maybe I'm missing something. But I was under the impression that ktfmt basically had 0 settings which makes it anti bike shedding (unlike ktlint 😓 ). 

So if the only difference between these formats is spacing... then why not just have ktfmt-2-space vs ktfmt-4-space.

If there are indeed more options other than spacing (that'd be news to me)... then yeah definitely having names that match a publicly defined spec would be awesome.

so:
1. kotlinlang style makes sense, because it has a public style defined here https://www.jetbrains.com/help/idea/code-style-kotlin.html
2. google-java-format style makes sense, because its here https://google.github.io/styleguide/javaguide.html

Is there a dropbox public style guide? Meta?

**Reply by JavierSegoviaCordoba** [reply]
Currently, there are at least three options (four if Dropbox is not deprecated/removed). So it is not possible to have only two options with 2 spaces, and 4 spaces.

Also, they are opinionated on those companies' standards. On the other hand, I wouldn't see it bad to show some hints in the IDE (tooltip? parenthesis indicating the number of spaces?).

**Reply by ColtonIdle** [reply]
Hm. Maybe is there a tldr of each option? I thought the two options were default/google (4 spaces) and dropbox (2 spaces) lol

**Reply by JavierSegoviaCordoba** [reply]
There are 4:

- Default (Facebook, I think it is 2 spaces?)
- Kotlinlang (4 spaces)
- Google (I think it is 2 spaces?)
- Dropbox (I think it is 2 spaces?)

I only use Kotlinlang, so I cannot confirm the amount of spaces other ones are using.

**Reply by ColtonIdle** [reply]
interesting... yeah. see this is the part that confuses me. lol. if 3 of the 4 are using 2 spaces. then clearly there are _other_ differences between those 3, right. so if thats the case having that outlined would be clutch... but also goes back to my original point that it seems like if my company wanted our own custom format, we could hypothethically configure that because there's probably a ktfmt config file for those 4 styles?

**Comment by hick209** [comment]
I'll propose this as part of the 1.0.0 roadmap.

Thanks for starting this discussion @JavierSegoviaCordoba

**Reply by hick209** [reply]
Proposed here https://github.com/facebook/ktfmt/discussions/451#discussioncomment-9305741

**Comment by hick209** [comment]
That will be the default as of version 1.0.0
